packages/core/src/agent/index.ts

```diff
@@ -776,6 +778,14 @@ export class Agent<
     };
   }
 
+  async generate<Z extends ZodSchema | JSONSchema7 | undefined = undefined>(
+    messages: string | string[] | CoreMessage[],
+    args?: AgentGenerateOptions<Z> & { output?: 'text'; experimental_output?: undefined },
+  ): Promise<GenerateTextResult<any, any>>;
+  async generate<Z extends ZodSchema | JSONSchema7 | undefined = undefined>(
+    messages: string | string[] | CoreMessage[],
+    args?: AgentGenerateOptions<Z> & ({ output: Z } | { experimental_output: Z }),
+  ): Promise<GenerateObjectResult<Z extends ZodSchema ? z.infer<Z> : any>>;
```

Comment:
This causes it to correctly infer the return object type as the input zod type. I left it as `any` for JSON schemas - no idea how to infer that properly

Reply:
shouldnt it be unknown instead of any?

Reply:
pushed up a commit, thanks!